packages/propeller/src/FeedSource.mts

```diff
@@ -2,32 +2,93 @@ import { Batch, Sink } from "./Types.js"
 import { ICheckpoints } from "./Checkpoints.js"
 import { sleep } from "./Sleep.js"
 
-export class TailingFeedSource {
+class CheckpointWriter {
   constructor(
-    private readonly sourceId: string,
-    private readonly tailSleepIntervalMs: number,
     private readonly groupName: string,
+    private readonly trancheId: string,
+    private readonly intervalMs: number,
     private readonly checkpoints: ICheckpoints,
-    private readonly sink: Sink,
-    private readonly crawl: (
-      trancheId: string,
-      position: bigint,
-      signal: AbortSignal,
-    ) => AsyncIterable<Batch>,
-    private readonly establishOrigin?: (tranche: string) => Promise<bigint>,
+    private committedPos: bigint,
+    private validatedPos = committedPos,
   ) {}
 
-  async start(trancheId: string, signal: AbortSignal) {
-    let pos = await this.checkpoints.load(this.groupName, trancheId, this.establishOrigin)
+  async start(signal: AbortSignal) {
+    while (!signal.aborted) {
+      await sleep(this.intervalMs, signal)
+      if (this.committedPos !== this.validatedPos) {
+        await this.checkpoints.commit(this.groupName, this.trancheId, this.validatedPos)
+        this.validatedPos = this.committedPos
```

Comment:
should this be the other way round ?
maybe one liners as to what these mean might help?
maybe call it writtenPos vs currentPos/confirmedPos ?
(in the .fs version, the committed is the value that's actually been written, which is probably a little confusing)
```
        log.Information("Ingester {partition} Ahead {activeReads}/{maxReads} Committed {committed} @ {validated}/{pos} ok {commits} failed {fails} Ingested {batches} ({streams:n0}s {events:n0}e) Cycles {cycles}",
                        partitionId, activeReads, maxReads, Option.toNullable committedEpoch, Option.toNullable validatedEpoch, Option.toNullable readEpoch, commits, commitFails, batchesPended, streamsPended, eventsPended, cycles)
```